plugins/home/src/components/VisitListener.tsx

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright 2023 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import React, {
+  createContext,
+  useState,
+  useEffect,
+  useContext,
+  Dispatch,
+  SetStateAction,
+  ReactNode,
+} from 'react';
+
+import { useLocation } from 'react-router-dom';
+
+import { visitsApiRef } from '../api';
+import { useApi } from '@backstage/core-plugin-api';
+import { stringifyEntityRef } from '@backstage/catalog-model';
+
+/** @public */
+export type VisitListenerContextValue = {
+  doNotTrack: boolean;
+  setDoNotTrack: Dispatch<SetStateAction<boolean>>;
+};
+
+const defaultVisitListenerContext: VisitListenerContextValue = {
+  doNotTrack: false,
+  setDoNotTrack: () => {},
+};
+
+/** @public */
+export const VisitListenerContext = createContext<VisitListenerContextValue>(
+  defaultVisitListenerContext,
+);
+
+/**
+ * @public
+ * This function returns an implementation of toEntityRef which is responsible
+ * for receiving a pathname and maybe returning an entityRef compatible with the
+ * catalog-model.
+ * By default this function uses the url root "/catalog" and the
+ * stringifyEntityRef implementation from catalog-model.
+ * Example:
+ *   const toEntityRef = getToEntityRef();
+ *   toEntityRef(\{ pathname: "/catalog/default/component/playback-order" \})
+ *   // returns "component:default/playback-order"
+ */
+export const getToEntityRef =
```

Comment:
Does it make sense to export this helper from the catalog plugin? Like [this](https://github.com/backstage/backstage/blob/54b411d9aed79d8e992a93cc9eaf0372eebd6e72/plugins/catalog/src/components/CatalogEntityPage/useEntityFromUrl.ts#L31) utility is.

Reply:
Agreed. Done in 8df35448d14ac54c23199c116cd97fe77014eb5f